synapse/storage/databases/main/push_rule.py

```diff
@@ -560,19 +560,19 @@ def _upsert_push_rule_txn(
         if isinstance(self.database_engine, PostgresEngine):
             sql = """
                 INSERT INTO push_rules_enable (id, user_name, rule_id, enabled)
-                VALUES (?, ?, ?, ?)
+                VALUES (?, ?, ?, TRUE)
                 ON CONFLICT DO NOTHING
             """
         elif isinstance(self.database_engine, Sqlite3Engine):
             sql = """
                 INSERT OR IGNORE INTO push_rules_enable (id, user_name, rule_id, enabled)
-                VALUES (?, ?, ?, ?)
+                VALUES (?, ?, ?, TRUE)
             """
         else:
             raise RuntimeError("Unknown database engine")
 
         new_enable_id = self._push_rules_enable_id_gen.get_next()
-        txn.execute(sql, (new_enable_id, user_id, rule_id, 1))
```

Comment:
Looks like the `enabled` column in the `push_rules` table is a `smallint` for some reason. I think we will need to revert this.

https://github.com/matrix-org/synapse/actions/runs/5527360927/jobs/10083858957?pr=15525#step:7:4215
```
  synapse_main |   File "/usr/local/lib/python3.11/site-packages/synapse/storage/database.py", line 760, in new_transaction
  synapse_main |     r = func(cursor, *args, **kwargs)
  synapse_main |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  synapse_main |   File "/usr/local/lib/python3.11/site-packages/synapse/storage/databases/main/push_rule.py", line 484, in _add_push_rule_highest_priority_txn
  synapse_main |     self._upsert_push_rule_txn(
  synapse_main |   File "/usr/local/lib/python3.11/site-packages/synapse/storage/databases/main/push_rule.py", line 575, in _upsert_push_rule_txn
  synapse_main |     txn.execute(sql, (new_enable_id, user_id, rule_id))
  synapse_main |   File "/usr/local/lib/python3.11/site-packages/synapse/storage/database.py", line 419, in execute
  synapse_main |     self._do_execute(self.txn.execute, sql, parameters)
  synapse_main |   File "/usr/local/lib/python3.11/site-packages/synapse/storage/database.py", line 471, in _do_execute
  synapse_main |     return func(sql, *args, **kwargs)
  synapse_main |            ^^^^^^^^^^^^^^^^^^^^^^^^^^
  synapse_main | psycopg2.errors.DatatypeMismatch: column "enabled" is of type smallint but expression is of type boolean
  synapse_main | LINE 3: ...  VALUES (2, '@alice:hs1', 'global/sender/@alice:hs1', TRUE)
  synapse_main |                                                                   ^
  synapse_main | HINT:  You will need to rewrite or cast the expression.
```

Reply:
We can probably inline the `1` still though. 🤷 

We used to use `smallint` instead of `bool` due to sqlite support.